canopen/node/local.py

```diff
@@ -11,16 +13,35 @@ def __init__(self, node_id, object_dictionary):
         self.callbacks = []
 
         self.sdo = SdoServer(0x600 + self.id, 0x580 + self.id, self)
+        self.nmt = NmtSlave(self.id, self)
+        self.add_callback(self._producer_hearbeat_time_callback)
 
     def associate_network(self, network):
         self.network = network
         self.sdo.network = network
+        self.nmt.network = network
         network.subscribe(self.sdo.rx_cobid, self.sdo.on_request)
+        network.subscribe(0, self.nmt.on_command)
```

Comment:
I only added support for having one callback per CAN-ID, so this will only work for one node. You need to change the callback implementation to allow multiple callbacks.

Reply:
I changed the implementation to use specific nmt command subscribe function. This should then work with multiple nodes.

This is the last change that you requested. I.e. this is my last update, if you don't request more changes.